src/Adapter/MSTest.TestAdapter/Execution/TypeCache.cs

```diff
@@ -810,7 +810,8 @@ private MethodInfo GetMethodInfoForTestMethod(TestMethod testMethod, TestClassIn
         else if (methodBase != null)
         {
             Type[] parameters = methodBase.GetParameters().Select(i => i.ParameterType).ToArray();
-            testMethodInfo = PlatformServiceProvider.Instance.ReflectionOperations.GetRuntimeMethod(methodBase.DeclaringType!, methodBase.Name, parameters);
+            // TODO: Should we pass true for includeNonPublic?
+            testMethodInfo = PlatformServiceProvider.Instance.ReflectionOperations.GetRuntimeMethod(methodBase.DeclaringType!, methodBase.Name, parameters, includeNonPublic: false);
```

Comment:
I'd say no except if after the refactoring we no longer support internal test methods (I hope we have a test for that).

Please remove the comment or fix the code.

Reply:
`includeNonPublic: false` preserves the existing behavior. But I'm suspecting that to be a bug when `DiscoverInternals` is used, which is why I left it as "TODO". Let's discuss/investigate more on Monday.

Reply:
Moving on by merging, I am creating a follow-up task #4508